Psychtoolbox/PsychCal/MeasureDpi.m

```diff
@@ -16,31 +19,54 @@
 %             both Mac and Windows
 % 11/6/06 dgp Updated from PTB-2 to PTB-3.
 % 01/31/16 mk Fix some wrong assumptions for PTB-3. Get rid of GetChar.
+% 8/13/19 mgg Mofify to use it with a credit card (common object of a
+%             known size).
+% 9/10/19 mgg Optional output: Withdraw the presets from the screen
+%             inc. the screen size, resolution, dot pitch and distance independent dpi.
 
-if nargin>1 || nargout>1
-    error('Usage: dpi=MeasureDpi(screen)');
+if nargin>2 || nargout>2
+    error([datestr(now),':>> Error wrong call to function, use it as: dpi=MeasureDpi(Screen)', newline, 'or [dpi, SP]=MeasureDpi(Screen, CC)']);
 end
-if nargin<1
-    theScreen=0;
+if isempty(theScreen)|| ~exist(theScreen)
+    theScreen=max(Screen('Screens')); %Use the second screen, most of the cases.
+    disp([datestr(now),':>> No screen value has been provided so I assume that you would like to use the secondary one...']);
 end
 AssertOpenGL;
+
 try
-    disp('Please find an object of known width to hold against the display.');
-    disp('E.g. a ruler or an 8.5-inch page.')
-    inches=input('Do you prefer inches (1) or cm (0)? ');
-    if inches
+    if nargin>1 && isequal(CC,1) % Using a credit card
         unitInches=1;
         units='inches'; % e.g. distance in inches
         unit='inch';    % e.g. 5 inch object
+        objectInches= 8.56*(1/2.54); % ID-1 format, ISO/IEC 7810 credit card (85.60 × 53.98 mm).
+        inches=input('Do you prefer inches (1) or cm (0)? ');
+        if inches
+            unitInches=1;
+            units='inches'; % e.g. distance in inches
+            unit='inch';    % e.g. 5 inch object
+        else
+            unitInches=1/2.54;
+            units='cm';
+            unit='cm';
+        end
```

Comment:
This should be moved before/outside the if statement for credit card, so you only need it once, not twice for the if and else branch.